### ISSUE TITLE
Improve Library Import

### ISSUE THREAD

**Issue by rgecko** [Casvt/Kapowarr#127]
**Is your feature request related to a problem? Please describe.**
Currently library import looks to import based on single issues, clicking import lists the first 20 comics.  There is no option for importing at the folder level and ignoring the individual issues in the folders

**Describe the solution you'd like**
have the option to import and scan based on files in folders OR based on the folder name itself.
if you import by folders a scan in the folder should run to validate if the files are in the folder that match to issues.
Further you could run batch jobs over time that don't spam comic vine to scrape the individual issue details post folder import.

**Comment by Casvt** [comment]
The folder limit is for the folders that the issues are in. So 20 would give 20 volumes. It would only give 20 issues in the scenario that each issue has a separate sub-folder. So I guess I could add an option to limit parent folders instead of folders, for when each issue is in their own sub-folder.

Matching purely based on folder is also dangerous because it's not guaranteed that all files inside are for that volume. And we'd have to fetch the issues, which we currently don't have to.

Kapowarr already groups files when searching, to reduce the amount of requests. And when viewing the proposal, you can edit the match for one file but also for all volume files in one go.

So basically, you're describing a problem that isn't there, except that you get 20 issues instead of 20 volumes because you probably have sub-folders for each issue. Why not set the limit to the highest (500?) until I add the option to limit one folder higher?

**Comment by rgecko** [comment]
The way I have things setup is Publisher/Comic Series (year)/comic series (year published).cbx 
when I look to import the list may list the first part of the issues as the first volume, and then later issues as the second volume, my assumption was this is trigged as I have individual issues by year published instead of the first year published, but maybe that is the wrong assumption then.

the other reason I thought a folder option would help is importing some series (trade paper backs seems to be the common thing), isn't detecting the file already in the folder that was imported once it's imported so it's trying to re-import again instead of ignoring that the folder is there, so eventually the list gets full unless I do some manual formatting to the file in the folder it's not detecting.

**Comment by Casvt** [comment]
> The way I have things setup is Publisher/Comic Series (year)/comic series (year published).cbx
when I look to import the list may list the first part of the issues as the first volume, and then later issues as the second volume, my assumption was this is trigged as I have individual issues by year published instead of the first year published, but maybe that is the wrong assumption then.

No it's true I've experienced this too. But I should just make sure that it grabs the year from the folder and not the filename and it should be fixed.

> the other reason I thought a folder option would help is importing some series (trade paper backs seems to be the common thing), isn't detecting the file already in the folder that was imported once it's imported so it's trying to re-import again instead of ignoring that the folder is there, so eventually the list gets full unless I do some manual formatting to the file in the folder it's not detecting.

The file should just be detected. Could you give an example of the filename and the matching volume (title+year or CV ID)?

**Comment by rgecko** [comment]
for the file problem a couple examples are "Monster (2021)" and "Hip Flask (2002)"  both of which are TPBs and correctly recognized.  the file format in the folder is: "Monsters 001 (2002).cbz" and "Hip Flask 001 (2002).cbz".  If I edit them to "Monsters (2021) Volume 01.cbz" it shows up and sees the file.   regular series all work just fine.

In other Arr's there's the option to basically "match files" and you can pick other files in the folder to force a match.

**Comment by Casvt** [comment]
Yes that is correct behaviour and I'm not going to change that. Files are only matched to TPB releases when they explicitly say 'tpb' or when they only mention the volume number and no issue number.

The manual file matching is planned: #46
